dist/images/ovndb-raft-functions.sh

```diff
@@ -238,28 +273,69 @@ ovsdb-raft() {
     ovn_raft_conn_ip_url_suffix=":[::]"
   fi
 
-  if [[ "${POD_NAME}" == "ovnkube-db-0" ]]; then
-    run_as_ovs_user_if_needed \
+  initial_raft_create=true
+  if [[ "${initialize}" == "true" ]]; then
+    # check to see if a cluster already exists. If it does, just join it.
+    counter=0
+    cluster_found=false
+    while [ $counter -lt 5 ]; do
+      if cluster_exists ${db} ${port}; then
+        cluster_found=true
+        break
+      fi
+      sleep 1
+      counter=$((counter+1))
+    done
+
+    if ${cluster_found}; then
+      echo "Cluster already exists for DB: ${db}"
+      initial_raft_create=false
+      run_as_ovs_user_if_needed \
       ${OVNCTL_PATH} run_${db}_ovsdb --no-monitor \
-      --db-${db}-cluster-local-addr=[${ovn_db_host}] \
-      --db-${db}-cluster-local-port=${raft_port} \
-      --db-${db}-cluster-local-proto=${transport} \
+      --db-${db}-cluster-local-addr=$(bracketify ${ovn_db_host}) --db-${db}-cluster-remote-addr=${init_ip} \
+      --db-${db}-cluster-local-port=${raft_port} --db-${db}-cluster-remote-port=${raft_port} \
+      --db-${db}-cluster-local-proto=${transport} --db-${db}-cluster-remote-proto=${transport} \
       ${db_ssl_opts} \
       --ovn-${db}-log="${ovn_loglevel_db}" &
-  else
-    # join the remote cluster node if the DB is not created
-    if [[ "${initialize}" == "true" ]]; then
-      wait_for_event ready_to_join_cluster ${db} ${port}
+    else
+      # either we need to initialize a new cluster or wait for db-0 to create it
+      if [[ "${POD_NAME}" == "ovnkube-db-0" ]]; then
+        echo "Cluster does not exist for DB: ${db}, creating new raft cluster"
+        run_as_ovs_user_if_needed \
+        ${OVNCTL_PATH} run_${db}_ovsdb --no-monitor \
+        --db-${db}-cluster-local-addr=$(bracketify ${ovn_db_host}) \
+        --db-${db}-cluster-local-port=${raft_port} \
+        --db-${db}-cluster-local-proto=${transport} \
+        ${db_ssl_opts} \
+        --ovn-${db}-log="${ovn_loglevel_db}" &
+      else
+        echo "Cluster does not exist for DB: ${db}, waiting for ovnkube-db-0 pod to create it"
+        wait_for_event cluster_exists ${db} ${port}
+        run_as_ovs_user_if_needed \
+        ${OVNCTL_PATH} run_${db}_ovsdb --no-monitor \
+        --db-${db}-cluster-local-addr=$(bracketify ${ovn_db_host}) --db-${db}-cluster-remote-addr=${init_ip} \
+        --db-${db}-cluster-local-port=${raft_port} --db-${db}-cluster-remote-port=${raft_port} \
+        --db-${db}-cluster-local-proto=${transport} --db-${db}-cluster-remote-proto=${transport} \
+        ${db_ssl_opts} \
+        --ovn-${db}-log="${ovn_loglevel_db}" &
+      fi
     fi
-    run_as_ovs_user_if_needed \
+  else
+    if cluster_exists ${db} ${port}; then
```

Comment:
This is to get the init_ip set again before trying to join the cluster.

Reply:
Yeah I don't think this check is necessary @abhat If initialize is false, that means db is not standalone and already exists, so we know its clustered.

Reply:
Right, I can remove the check, but essentially we need to still call cluster_exists to set the init_ip.

Reply:
I didn't see your comment @abhat before I responded. Yeah we need cluster_exists to get the init_ip. I think the check is still valid because if the cluster exists we should be able to get the leader IP. If we cant then something is wrong and I dont think we should assume its db-0 pod.

Reply:
no, we don't need cluster_exists check, init_ip is actually not used in this case. it is only used for the initial creation of cluster DB to join the cluster. But I am not sure how to use the wrapup ovn-ctl command to simply start ovsdb-server without specifying cluster-remote-addr

Reply:
figured with @cathy-zhou 's help that we don't need the remote-addr and remote-port at all. So long as we pass the nb-cluster-local-addr/sb-cluster-local-addr, ovn-ctl will start the ovsdb-server as a part of the existing cluster since the remote information is saved in the db already.

Reply:
great